src/bundle/Controller/ContentController.php

```diff
@@ -157,4 +156,62 @@ public function createDraftAction(Request $request): Response
 
         return $this->redirectToRoute('ezplatform.dashboard');
     }
+
+    /**
+     * @param Request $request
+     *
+     * @return Response
+     *
+     * @throws \InvalidArgumentException
+     * @throws TranslationInvalidArgumentException
+     * @throws APIRepositoryInvalidArgumentException
+     * @throws UnauthorizedException
+     * @throws InvalidOptionsException
+     */
+    public function setMainLocationAction(Request $request): Response
+    {
+        $form = $this->formFactory->updateContentMainLocation();
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            $result = $this->submitHandler->handle($form, function (ContentUpdateMainLocationData $data) {
+                $contentInfo = $data->getContentInfo();
+                $contentMetadataUpdateStruct = new ContentMetadataUpdateStruct();
```

Comment:
As far as I remember we use mappers to do the conversion between repository structs and our DTOs? @Nattfarinn am I right?

Reply:
I will check this and fix if needed

Reply:
Mapper is overkill and will be misleading. Revert it.